packages/composite-checkout-wpcom/src/components/form-field-annotation.tsx

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import React, { FunctionComponent } from 'react';
+import styled from '@emotion/styled';
+
+type FormFieldAnnotationProps = {
+	// Semantic props
+	labelText: string;
+	descriptionText?: string;
+	errorMessage: string;
+
+	// Functional props
+	isError: boolean;
+	isDisabled: boolean;
+
+	// Technical props
+	className?: string;
+	labelId: string;
+	descriptionId: string;
+	formFieldId: string;
```

Comment:
Are the ID props really always mandatory? If I don't provide `labelText`, then the label is not rendered and `labelId` and `formFieldId` are not used.

Reply:
I decided to make `labelText` required, and so `labelId` and `formFieldId` are also required as they're used for aria attributes. The `errorMessage` is required since that's the purpose of this component, and `descriptionId` is used for that. The names don't make this obvious though, so I changed `descriptionText` to `normalDescription` and `errorMessage` to `errorDescription`.